stripe/http_client.py

```diff
@@ -13,23 +13,27 @@
 from stripe import error, util
 from stripe.request_metrics import RequestMetrics
 
+from typing import NoReturn, Optional, cast
+from typing_extensions import TypedDict
```

Comment:
We need to figure out a pattern for when we import from typing_extensions or typing

Reply:
Let's do typing_extensions everywhere